pandas/tests/indexes/test_base.py

```diff
@@ -2103,3 +2103,24 @@ def test_intersect_str_dates(self):
         res = i2.intersection(i1)
 
         assert len(res) == 0
+
+    def test_searchsorted_monotonic(self):
+        # GH17271
+        idx_inc = Index([0, 2, 4])
+        idx_dec = Index([4, 2, 0])
```

Comment:
would be better to move these to ``common.py/Base`` as they be tested generically.